From e53f834829a4fc445e81c42eee914912ec07fa3a Mon Sep 17 00:00:00 2001
From: Jonas Claes <jonas@jonasclaes.be>
Date: Tue, 26 Oct 2021 17:19:35 +0200
Subject: [PATCH] Added setting of a custom host name.

---
 src/Dhcp.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Dhcp.cpp b/src/Dhcp.cpp
index cc71c68..3ada5ee 100644
--- a/src/Dhcp.cpp
+++ b/src/Dhcp.cpp
@@ -204,12 +204,17 @@ void DhcpClass::send_DHCP_MESSAGE(uint8_t messageType, uint16_t secondsElapsed)
 
     // OPT - host name
     buffer[16] = hostName;
+    #ifndef CUSTOM_HOST_NAME
     buffer[17] = strlen(HOST_NAME) + 6; // length of hostname + last 3 bytes of mac address
     strcpy((char*)&(buffer[18]), HOST_NAME);
 
     printByte((char*)&(buffer[24]), _dhcpMacAddr[3]);
     printByte((char*)&(buffer[26]), _dhcpMacAddr[4]);
     printByte((char*)&(buffer[28]), _dhcpMacAddr[5]);
+    #else
+    buffer[17] = strlen(CUSTOM_HOST_NAME);
+    strcpy((char*)&(buffer[18]), CUSTOM_HOST_NAME);
+    #endif
 
     //put data in W5100 transmit buffer
     _dhcpUdpSocket.write(buffer, 30);
-- 
2.31.1

